code/debris/debris.cpp

```diff
@@ -460,15 +460,15 @@ object *debris_create(object *source_obj, int model_num, int submodel_num, vec3d
 	if (!hull_flag) {
 		if (model_num >= 0) {
 			db->model_num = model_num;
-			db->submodel_num = (myrand() >> 4) % sip->generic_debris_num_submodels;
+			db->submodel_num = (util::Random::next() >> 4) % sip->generic_debris_num_submodels;
 		}
 		else if (vaporize) {
 			db->model_num = Debris_vaporize_model;
-			db->submodel_num = (myrand() >> 4) % Debris_num_submodels;
+			db->submodel_num = (util::Random::next() >> 4) % Debris_num_submodels;
 		}
 		else {
 			db->model_num = Debris_model;
-			db->submodel_num = (myrand() >> 4) % Debris_num_submodels;
+			db->submodel_num = (util::Random::next() >> 4) % Debris_num_submodels;
```

Comment:
These three cases look suspiciously like they depend on the old maximum value. These could be changed to make use of the range limiting built into `Random::next()`.

Reply:
It's well-known C folklore that `rand()` has historically been a lot less random in its low-order bits than it should be, so I assume the bitshift here is simply to chop them out before clamping into a range with modulo. I agree this should be replaced with `util::Random::next(Debris_num_submodels)`.

Reply:
I agree with @PhantomHoover  's explanation as the most likely one, in which case I also agree that we can get rid of these right shifts. They appear in a number of places in the code.

For all I know, MT may also be less random in its low order bits, but IMO even if so, the extra bit twiddling isn't worth it.